### PR TITLE
Rename hie-core to ghcide

### PR DESCRIPTION
**Pull request by cocreature**
The name `hie-core` has caused a lot of confusion as to how we relate
to haskell-ide-engine so changing it should hopefully help with that.
I also think that ghcide is still a good name once we hopefully
integrate with haskell-ide-engine more closely.

The name ghcide seems to have a reasonable amount of support on
Twitter https://twitter.com/ndm_haskell/status/1170681262987710464
which is of course the only good way to come up with names.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by ndmitchell** [comment]
Could we leave a README in hie-core saying where it's gone, since I'm conscious I just put out a talk with a big URL that lots of people are seemingly following that points to somewhere that is now going to 404.

**Comment by cocreature** [comment]
Added a readme

File: compiler/ghcide/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# `ghcide` (Haskell IDE engine)
```
Comment (hurryabit) [inline compiler/ghcide/README.md]: I suppose we want to remove the parts in parentheses.

File: compiler/ghcide/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+There are more details about our approach [in this blog post](https://4ta.uk/p/shaking-up-the-ide).
+
+## Using it
+
+### Install `ghcide`
+
+#### With Nix
+
+[See ghcide-nix repository](https://github.com/hercules-ci/ghcide-nix)
```
Comment (hurryabit) [inline compiler/ghcide/README.md]: That doesn't work.

File: compiler/hie-core/README.md [hunk 3]
```diff
@@ -1,125 +1,4 @@
...
-```vim
-au User lsp_setup call lsp#register_server({
-    \ 'name': 'hie-core',
-    \ 'cmd': {server_info->['/your/path/to/hie-core', '--lsp']},
-    \ 'whitelist': ['haskell'],
-    \ })
-```
-
-To verify it works move your cursor over a symbol and run `:LspHover`.
+`hie-core` has been renamed to `ghcide`. See `compiler/ghcide` for the
```
Comment (hurryabit) [inline compiler/hie-core/README.md]: Could we turn the reference into a proper clickable link?

**Pull request merged**